examples/bpi-monolith-test-suite/bpi/components/zkmock/blockchain/shieldUtils.js

```diff
@@ -0,0 +1,83 @@
+import {
+  connectToBlockchain,
+  compileContract,
+  deployContract,
+  getDeployedShieldContract,
+} from "./blockchain";
+import { fs } from "fs";
+
+const URL = "https://localhost:8080";
+
+export const deployShield = async () => {
+  var web3 = await connectToBlockchain();
+
+  //compile and deploy Verifier contract
+  var verifierContractAbsolutePath = path.resolve(
+    "../../zkcircuits/ordercircuit/order.sol"
+  );
+  var compiledVerifierContract = compileContract(
+    "Verifier",
+    verifierContractAbsolutePath
+  );
+  var deployedVerifierContract = deployContract(
+    compiledVerifierContract,
+    web3.eth.defaultAccount
+  );
+
+  //compile and deploy Shield contract
+  var shieldContractAbsolutePath = path.resolve("./contracts/Shield.sol");
+  var compiledShieldContract = compileContract(
+    "Shield",
+    shieldContractAbsolutePath,
+    [deployedVerifierContract.options.address]
+  );
+  var deployedShieldContract = deployContract(
+    compiledShieldContract,
+    web3.eth.defaultAccount
+  );
+
+  //Store the compiled Shield contract to access it later.
+  fs.writeFile("./contracts/Shield.json", compiledShieldContract, (err) => {
+    if (err) {
+      console.error(err);
+      return;
+    }
+    //file written successfully
+  });
+};
+
+export const verifyAndAddNewLeaf = async (a, b, c, input, _newCommitment) => {
+  var web3 = await connectToBlockchain();
+
+  //Connect to deployed Shield Contract
+  var shieldContract = await getDeployedShieldContract();
+  var shieldContractAbi = shieldContract.abi;
+  var shield = new web3.eth.Contract(
+    shieldContractAbi,
+    web3.eth.defaultAccount
```

Comment:
shouldn't this argument be address of deployed smart contract?

Reply:
Updated the code to reflect the shield contract address.